persist/pg.go

```diff
@@ -22,10 +25,47 @@ func NewPgStore(databaseUrl string) (*PgStore, error) {
 	return &PgStore{pool: pool}, nil
 }
 
-func (s *PgStore) Set(ctx context.Context, webhook Webhook) error {
+func (s *PgStore) Set(ctx context.Context, webhook Webhook) (*Webhook, error) {
 	pk, err := hex.DecodeString(webhook.Pubkey)
 	if err != nil {
-		return err
+		return nil, err
+	}
+	pubkeyUsername, err := s.getPubkeyUsername(ctx, pk)
+	if err != nil {
+		return nil, err
+	}
+	if webhook.Username != nil {
+		// The set request includes a username. Insert the username for the pubkey if no record
+		// was found, otherwise update the pubkey's record with the new username. 
+		// If another record already uses this username, there will be an error returned.
+		username := strings.ToLower(*webhook.Username)
+		var res pgconn.CommandTag
+		if pubkeyUsername == nil {
+			res, err = s.pool.Exec(
+				ctx,
+				`INSERT INTO public.lnurl_pubkey_usernames (pubkey, username) values ($1, $2)`,
```

Comment:
Maybe do an `ON CONFLICT (pubkey) DO UPDATE SET username = $2` in case of a race between `getPubkeyUsername` and this insert. Actually, thinking about it, that might remove the need for the `getPubkeyUsername` call and the `UPDATE` call below?